build-system/pr-check/esm-tests.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script builds and tests the AMP runtime in esm mode.
+ * This is run during the CI stage = test; job = esm tests.
+ */
+
+const colors = require('ansi-colors');
+const {
+  printChangeSummary,
+  startTimer,
+  stopTimer,
+  timedExecOrDie: timedExecOrDieBase,
+  downloadEsmDistOutput,
+  downloadDistOutput,
+} = require('./utils');
+const {determineBuildTargets} = require('./build-targets');
+const {isTravisPullRequestBuild} = require('../common/travis');
+
+const FILENAME = 'esm-tests.js';
+const FILELOGPREFIX = colors.bold(colors.yellow(`${FILENAME}:`));
+const timedExecOrDie = (cmd, unusedFileName) =>
+  timedExecOrDieBase(cmd, FILENAME);
+
+function main() {
+  const startTime = startTimer(FILENAME, FILENAME);
+
+  if (!isTravisPullRequestBuild()) {
+    downloadDistOutput(FILENAME);
```

Comment:
@rsimha need your opinion here. the ESM tests need the 3p code (`f.js` specifically) from normal build. This is because the script for these "remote.html" tests should always use `f.js` instead of `f.mjs` because these files are hosted by the publisher so we have no control (currently) to enforce them to use `f.mjs`.

so i'd like to download the `f.js` files from the dist output zip, should i go an download the whole zip file or should i try and break up the nomodule 3p code into its own zip archive?

Reply:
I like the first option you proposed (download the entire set of nomodule build artifacts and unzip it to the same `dist` folder). With this, if there are more tests in future that need to test other module <-> nomodule switchover codepaths, they'll work without additional infra changes.